src/extension-support/extension-manager.js

```diff
@@ -389,6 +393,34 @@ class ExtensionManager {
         return menuItems;
     }
 
+    /**
+     * Ready the custom serialization and deserialization information in an extension's metadata.
+     * Before: map of short opcode (like `variable`) to `serialize` & `deserialize` function names.
+     * After: map of extended opcode (like `data_variable`) to synchronous callables for `serialize` and `deserialize`.
+     * @param {string} extensionId - the ID of the extension, like 'data'.
+     * @param {string} serviceName - the name of the service registered for this extension.
+     * @param {object.<object.<string>>} rawSerializationInfo - map of block opcode to `serialize` & `deserialize` info.
+     * @return {object.<object.<function>>} - prepared serialization info as described above.
+     */
+    _prepareSerializationInfo (extensionId, serviceName, rawSerializationInfo) {
+        rawSerializationInfo = rawSerializationInfo || {};
+        const cookedSerializationInfo = {};
+        for (const opcode in rawSerializationInfo) {
+            if (!rawSerializationInfo.hasOwnProperty(opcode)) continue;
+            const functionNames = rawSerializationInfo[opcode];
+            const extendedOpcode = `${extensionId}_${opcode}`;
+            const boundFunctions = {};
+            if (functionNames.serialize) {
```

Comment:
Do we want to throw an error or log a warning if `serialize` or `deserialize` are not specified in the serialize descriptor for the opcode?